pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/ReplicaGroupInstanceSelector.java

```diff
@@ -135,37 +137,31 @@ private Pair<Map<String, String>, Map<String, String>> selectServersUsingRoundRo
   }
 
   private Pair<Map<String, String>, Map<String, String>> selectServersUsingAdaptiveServerSelector(List<String> segments,
-      int requestId, SegmentStates segmentStates, List<String> serverRankList) {
+      int requestId, SegmentStates segmentStates, Map<String, Integer> serverRankMap) {
     Map<String, String> segmentToSelectedInstanceMap = new HashMap<>(HashUtil.getHashMapCapacity(segments.size()));
     // No need to adjust this map per total segment numbers, as optional segments should be empty most of the time.
     Map<String, String> optionalSegmentToInstanceMap = new HashMap<>();
-    for (String segment : segments) {
+    segments.forEach(segment -> {
```

Comment:
nit: while smaller lambdas are good to simplify the code, I think a multiline block here might complicate the stack trace if some error is thrown inside. I personally think we can just use for loop here.

Reply:
Sure switched back to for